Refactor using React & Typescript

krishnan-r opened the pull request.

diocas reviewed (commented):
So... this is almost a complete rewrite of the extension (in ts)? Nice!
When you said rewrite in TS I though it was only the lab extension, but reusing components makes total sense.
I didn't have the time to check the ts files, only the others, so this is a half review. I decided to submit it since I'm on holidays and won't be able to finish until 1 week from now.

I like the organization of the different extensions.
Please gitignore the jar outputs, as they were commited and shouldn't. You also don't build the jars when building the extension. Are these decisions on purpose?

~~Side note: what are we going to do with `swan_spark_conf`, which is the variable our users currently use?~~ nvm!

krishnan-r commented on sparkmonitor/__init__.py:
```diff
@@ -18,11 +19,15 @@ def _jupyter_nbextension_paths():
 
+HERE = Path(__file__).parent.resolve()
 
-def _jupyter_server_extension_paths():
```
Yes, I've removed the server extension that does the proxying entirely.

It is currently disabled in the code in the production version we have as it only works for local spark. 

I will explore using https://github.com/jupyterhub/jupyter-server-proxy or a more generic solution to proxy things running inside the session, or if not I will add it back with the necessary changes as a separate PR.

diocas replied:
Ok! 
I have some doubts `jupyter-server-proxy` does what we want, since I believe it doesn't rewrite the content to update the urls and only translates a url to a port and nothing else. But check it ofc.

krishnan-r commented on sparkmonitor/kernelextension.py:
```diff
@@ -153,17 +154,10 @@ def load_ipython_extension(ipython):
     global ip, monitor  # For Debugging
```
This was like this in the past. I will refactor this in a separate PR after this one is merged.

diocas replied:
Fine by me!

krishnan-r commented:
Thanks for the review @diocas, 
- I've removed the JARs from git, and added the commands to build them through setup.py and `yarn run build:prod`
- I've added a CI for linting the frontend code, will do the backend as a separate PR
- addressed the other comments
